components/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+PROJECT(components CXX)
+
+FILE(GLOB COMP_SOURCES vfs/*.h vfs/*.hpp vfs/*.cpp
+	archives/*.h archives/*.hpp archives/*.cpp
+)
+
+IF (WIN32)
+	FILE(GLOB COMP_MISC_SOURCES misc/*.h misc/*.hpp misc/*.cpp)
```

Comment:
OOps, there's a bug here. Sorry about that.
It should be COMP_WIN32_SOURCES, which is used line 11.